### PR TITLE
Update composer.json

### PR DESCRIPTION
**Pull request by bobdenotter**

### CODE REVIEW

**Comment by JarJak** [comment]
Build fails probably due to api platform update.

**Comment by bobdenotter** [comment]
Yes, looks like it.. I'd assumed that `2.4.2` would work where our previous dependency was `2.4.0-beta`, but I guess not. :-)

**Comment by JarJak** [comment]
@bobdenotter can you check which was the last working version of api platform?

**Comment by bobdenotter** [comment]
Output 
Using curl, with `2.4.0-beta.1`, i get this: 

```
{
  "@context": "/api/contexts/Content",
  "@id": "/api/contents",
  "@type": "hydra:Collection",
  "hydra:member": [
    {
      "@id": "/api/contents/1",
      "@type": "Content",
      "id": 1,
      "contentType": "homepage",
      "modifiedAt": "2018-10-01T22:49:28+02:00",
      "publishedAt": "2018-10-18T11:33:47+02:00",
      "fieldValues": {
        "title": "Porro quibusdam corrupti assumenda soluta earum consequatur quo.",
        "image": {
          "filename": "kitten.jpg",
          "alt": "A cute kitten",
          "path": "/thumbs/kitten.jpg?w=240&h=160&area=files"
        },
        "teaser": "Quam corrupti omnis ratione necessitatibus ipsam aut. Rem eum sequi aut sed quia doloribus illo. Minus laudantium beatae expedita nostrum sint.\n\nAt commodi eius voluptates voluptas id qui velit. Aut aut nobis hic aspernatur. Consequuntur ipsa et nostrum laborum tempore tempore. Odio dolores consectetur quis officia provident aut. Ad quis tenetur necessitatibus soluta.\n\nExercitationem sunt saepe debitis laudantium. Praesentium quasi animi non nisi. Beatae sed repudiandae quo. Expedita illo quos aperiam sunt modi quia. Natus velit quidem sit quia.",
        "content": "Voluptatem nobis veniam eum voluptatem soluta. Suscipit voluptatem quia et nesciunt. Voluptatum repudiandae aut officiis asperiores et consequatur.\n\nIste aspernatur minus commodi. Nobis et mollitia itaque in suscipit. Distinctio sunt maxime iste iusto. Ipsum ea id dolorem omnis.\n\nSunt voluptas facilis itaque placeat impedit commodi. Laborum pariatur eligendi omnis est explicabo. Et ut soluta et molestiae.",
        "contentlink": "Tempora qui id provident rerum vel.",
        "slug": "porro-quibusdam-corrupti-assumenda-soluta-earum-consequatur-quo"
      },
      "taxonomyValues": {
        "categories": {
          "events": "Events",
          "movies": "Movies"
        },
        "tags": {
          "adult": "adult",
          "biology": "biology",
          "research": "research",
          "space": "space"
        }
      },
      "authorName": "Admin",
      "extras": {
        "title": "Porro quibusdam corrupti assumenda soluta earum consequatur quo.",
        "image": {
          "filename": "kitten.jpg",
          "alt": "A cute kitten",
          "path": "/thumbs/kitten.jpg?w=240&h=160&area=files"
        },
        "excerpt": "Porro quibusdam corrupti assumenda soluta earum consequatur quo. Quam corrupti omnis …",
        "link": "/homepage/porro-quibusdam-corrupti-assumenda-soluta-earum-consequatur-quo",
        "editLink": "/bolt/edit/1"
      }
    }
  ],
  "hydra:totalItems": 1,
  "hydra:view": {
    "@id": "/api/contents?id=1",
    "@type": "hydra:PartialCollectionView"
  },
  "hydra:search": {
    "@type": "hydra:IriTemplate",
    "hydra:template": "/api/contents{?id,id[],contentType,contentType[],status,status[],createdAt,createdAt[],modifiedAt,modifiedAt[],publishedAt,publishedAt[],depublishedAt,depublishedAt[]}",
    "hydra:variableRepresentation": "BasicRepresentation",
    "hydra:mapping": [
      {
        "@type": "IriTemplateMapping",
        "variable": "id",
        "property": "id",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "id[]",
        "property": "id",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "contentType",
        "property": "contentType",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "contentType[]",
        "property": "contentType",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "status",
        "property": "status",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "status[]",
        "property": "status",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "createdAt",
        "property": "createdAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "createdAt[]",
        "property": "createdAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "modifiedAt",
        "property": "modifiedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "modifiedAt[]",
        "property": "modifiedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "publishedAt",
        "property": "publishedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "publishedAt[]",
        "property": "publishedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "depublishedAt",
        "property": "depublishedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "depublishedAt[]",
        "property": "depublishedAt",
        "required": false
      }
    ]
  }
}
```

using `2.4.0`, i get this: 

```
{
  "@context": "/api/contexts/Content",
  "@id": "/api/contents",
  "@type": "hydra:Collection",
  "hydra:member": [
    {
      "id": 1,
      "contentType": "homepage",
      "slug": "porro-quibusdam-corrupti-assumenda-soluta-earum-consequatur-quo",
      "author": {
        "id": 1,
        "displayName": "Admin",
        "username": "admin",
        "email": "admin@example.org"
      },
      "fields": {
        "title": "Porro quibusdam corrupti assumenda soluta earum consequatur quo.",
        "image": {
          "filename": "kitten.jpg",
          "alt": "A cute kitten",
          "path": "/thumbs/kitten.jpg?w=240&h=160&area=files"
        },
        "teaser": "Quam corrupti omnis ratione necessitatibus ipsam aut. Rem eum sequi aut sed quia doloribus illo. Minus laudantium beatae expedita nostrum sint.\n\nAt commodi eius voluptates voluptas id qui velit. Aut aut nobis hic aspernatur. Consequuntur ipsa et nostrum laborum tempore tempore. Odio dolores consectetur quis officia provident aut. Ad quis tenetur necessitatibus soluta.\n\nExercitationem sunt saepe debitis laudantium. Praesentium quasi animi non nisi. Beatae sed repudiandae quo. Expedita illo quos aperiam sunt modi quia. Natus velit quidem sit quia.",
        "content": "Voluptatem nobis veniam eum voluptatem soluta. Suscipit voluptatem quia et nesciunt. Voluptatum repudiandae aut officiis asperiores et consequatur.\n\nIste aspernatur minus commodi. Nobis et mollitia itaque in suscipit. Distinctio sunt maxime iste iusto. Ipsum ea id dolorem omnis.\n\nSunt voluptas facilis itaque placeat impedit commodi. Laborum pariatur eligendi omnis est explicabo. Et ut soluta et molestiae.",
        "contentlink": "Tempora qui id provident rerum vel.",
        "slug": "porro-quibusdam-corrupti-assumenda-soluta-earum-consequatur-quo"
      },
      "taxonomies": {
        "categories": {
          "events": "Events",
          "movies": "Movies"
        },
        "tags": {
          "adult": "adult",
          "biology": "biology",
          "research": "research",
          "space": "space"
        }
      },
      "extras": {
        "title": "Porro quibusdam corrupti assumenda soluta earum consequatur quo.",
        "image": {
          "filename": "kitten.jpg",
          "alt": "A cute kitten",
          "path": "/thumbs/kitten.jpg?w=240&h=160&area=files"
        },
        "excerpt": "Porro quibusdam corrupti assumenda soluta earum consequatur quo. Quam corrupti omnis …",
        "link": "/homepage/porro-quibusdam-corrupti-assumenda-soluta-earum-consequatur-quo",
        "editLink": "/bolt/edit/1"
      },
      "status": "published",
      "icon": "fa-home",
      "createdAt": "2018-11-17T03:46:15+01:00",
      "modifiedAt": "2018-10-01T22:49:28+02:00",
      "publishedAt": "2018-10-18T11:33:47+02:00",
      "depublishedAt": "2018-11-27T18:29:31+01:00"
    }
  ],
  "hydra:totalItems": 1,
  "hydra:view": {
    "@id": "/api/contents?id=1",
    "@type": "hydra:PartialCollectionView"
  },
  "hydra:search": {
    "@type": "hydra:IriTemplate",
    "hydra:template": "/api/contents{?id,id[],contentType,contentType[],status,status[],createdAt,createdAt[],modifiedAt,modifiedAt[],publishedAt,publishedAt[],depublishedAt,depublishedAt[]}",
    "hydra:variableRepresentation": "BasicRepresentation",
    "hydra:mapping": [
      {
        "@type": "IriTemplateMapping",
        "variable": "id",
        "property": "id",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "id[]",
        "property": "id",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "contentType",
        "property": "contentType",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "contentType[]",
        "property": "contentType",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "status",
        "property": "status",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "status[]",
        "property": "status",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "createdAt",
        "property": "createdAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "createdAt[]",
        "property": "createdAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "modifiedAt",
        "property": "modifiedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "modifiedAt[]",
        "property": "modifiedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "publishedAt",
        "property": "publishedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "publishedAt[]",
        "property": "publishedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "depublishedAt",
        "property": "depublishedAt",
        "required": false
      },
      {
        "@type": "IriTemplateMapping",
        "variable": "depublishedAt[]",
        "property": "depublishedAt",
        "required": false
      }
    ]
  }
}
```

I've updated the test that was failing

File: tests/api/content/get_content.feature [hunk 1]
```diff
@@ -178,8 +178,8 @@ Feature: Get content with API
   """
   {
      "@context": "/api/contexts/Content",
-     "@id": "/api/contents/1",
-     "@type": "Content",
```
Comment (JarJak) [inline tests/api/content/get_content.feature]: that's strange, you shouldn't get a collection while asking for a single entity

**Comment by JarJak** [comment]
Related issue: https://github.com/api-platform/api-platform/issues/1018

**Comment by JarJak** [comment]
I'd vote on closing this. In composer.json we should keep lowest working minor versions of dependencies, because later when someone will use Bolt as a dependency and want to use any older package version it will conflicts. As long as we can update them with `composer update` I don't see any issue here.

That being said, there are some packages that should be updated. These are:
`"coduo/php-matcher": "^3.1.2"`
`"phpunit/phpunit": "^8.0"`
`"symplify/easy-coding-standard": "^5.4.14"`

**Pull request closed without merging**